### PR TITLE
sdk-backend-spi: Add API to fetch attribute dataset

### PR DESCRIPTION
**Pull request by xMort**
JIRA: BB-2970

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3875/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5508/ : SUCCESS in 15m 57s

**Comment by no23reason** [comment]
Please fix the commit body, the JIRA reference is invalid.

File: libs/sdk-backend-spi/src/workspace/attributes/index.ts [hunk 1]
```diff
@@ -64,4 +68,11 @@ export interface IWorkspaceAttributesService {
      * @returns promise returning batch of attribute references.
      */
     getCommonAttributesBatch(attributesRefsBatch: ObjRef[][]): Promise<ObjRef[][]>;
+
+    /**
+     * Get information about the given attribute's dataset
+     * @param ref - ref of the attribute
+     * @returns promise of metadata object
+     */
+    getAttributeDatasetMeta(ref: ObjRef): Promise<IMetadataObject>;
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/attributes/index.ts:77]: Do we have some more concrete type for the returned value? Also, to keep naming consistent, we should probably drop the "Meta" part, the other methods here also return metadata objects.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1409

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1135/ : SUCCESS in 55s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1405/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/146/ : SUCCESS in 6m 54s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5513/ : SUCCESS in 15m 31s

**Pull request merged**